delphi-frontend/src/main/java/au/com/integradev/delphi/reporting/DelphiIssueBuilder.java

```diff
@@ -18,218 +18,36 @@
  */
 package au.com.integradev.delphi.reporting;
 
-import au.com.integradev.delphi.check.MasterCheckRegistrar;
-import au.com.integradev.delphi.file.DelphiFile.DelphiInputFile;
-import com.google.common.base.Preconditions;
 import com.google.errorprone.annotations.FormatMethod;
 import com.google.errorprone.annotations.FormatString;
-import java.util.Arrays;
-import java.util.Collections;
 import java.util.List;
-import java.util.Optional;
-import java.util.stream.Collectors;
-import javax.annotation.Nullable;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-import org.sonar.api.batch.fs.InputFile;
-import org.sonar.api.batch.fs.TextPointer;
-import org.sonar.api.batch.fs.TextRange;
-import org.sonar.api.batch.sensor.SensorContext;
-import org.sonar.api.batch.sensor.issue.NewIssue;
-import org.sonar.api.batch.sensor.issue.NewIssueLocation;
-import org.sonar.api.rule.RuleKey;
-import org.sonar.api.rule.RuleScope;
 import org.sonar.plugins.communitydelphi.api.ast.DelphiNode;
-import org.sonar.plugins.communitydelphi.api.check.DelphiCheck;
 import org.sonar.plugins.communitydelphi.api.check.DelphiCheckContext;
-import org.sonar.plugins.communitydelphi.api.check.DelphiCheckContext.Location;
 import org.sonar.plugins.communitydelphi.api.check.FilePosition;
 
-/**
- * Based directly on {@code InternalJavaIssueBuilder} from the sonar-java project.
- *
- * @see <a
- *     href="https://github.com/SonarSource/sonar-java/blob/master/java-frontend/src/main/java/org/sonar/java/reporting/InternalJavaIssueBuilder.java">
- *     InternalJavaIssueBuilder </a>
- */
-public final class DelphiIssueBuilder {
-  private static final Logger LOG = LoggerFactory.getLogger(DelphiIssueBuilder.class);
-
-  private static final String MESSAGE_NAME = "message";
-  private static final String FLOWS_NAME = "flows";
-  private static final String SECONDARIES_NAME = "secondaries";
-
-  private final DelphiCheck check;
-  private final SensorContext context;
-  private final DelphiInputFile delphiFile;
-  private final MasterCheckRegistrar checkRegistrar;
-  private FilePosition position;
-  private String message;
-  @Nullable private List<DelphiCheckContext.Location> secondaries;
-  @Nullable private List<List<Location>> flows;
-  @Nullable private Integer cost;
-  private boolean reported;
-
-  public DelphiIssueBuilder(
-      DelphiCheck check,
-      SensorContext context,
-      DelphiInputFile delphiFile,
-      MasterCheckRegistrar checkRegistrar) {
-    this.check = check;
-    this.context = context;
-    this.delphiFile = delphiFile;
-    this.checkRegistrar = checkRegistrar;
-  }
-
-  private static void requiresValueToBeSet(Object target, String targetName) {
-    Preconditions.checkState(target != null, "A %s must be set first.", targetName);
-  }
+public interface DelphiIssueBuilder {
```

Comment:
Offhand, I don't think that custom rules plugins will be able to use quick-fixes with the API boundaries as they are currently drawn.

Something to consider:
* Should `DelphiIssueBuilder` be exposed through the custom rules API?
  * It's returned by `DelphiCheckContext::newIssue`, which makes me think yes.
* Should the `DelphiQuickFix`/`DelphiTextEdit` be exposed through the custom rules API?
  * Creating quick-fixes will require direct references to these classes, which makes me think yes.

Reply:
Yes, I agree. Where do these API boundaries get drawn? I assumed by putting it with `DelphiIssueBuilder`, they would be accessible.

Reply:
Anything in the `org.sonar.plugins.communitydelphi.api` package is exposed via the custom rules API.

Reply:
I've exposed everything now (I think). Worth double checking.